django/db/migrations/migration.py

```diff
@@ -162,8 +162,10 @@ def unapply(self, project_state, schema_editor, collect_sql=False):
                 schema_editor.collected_sql.append("--")
                 if not operation.reduces_to_sql:
                     continue
-            if not schema_editor.connection.features.can_rollback_ddl and operation.atomic:
-                # We're forcing a transaction on a non-transactional-DDL backend
+            atomic_operation = operation.atomic or (self.atomic and operation.atomic is not False)
```

Comment:
Thanks to @timgraham for helping me in IRC to get my test environment set up correctly (I had neglected to specify an `other` databse in addition to the `default` database - both are required for the tests to run).

As for @ramiro's suggestion: this would not adhere to the behavior [specified](https://docs.djangoproject.com/en/1.11/ref/migration-operations/#runpython) for `RunPython` operations.  Specifically:
>By default, RunPython will run its contents inside a transaction on databases that do not support DDL transactions (for example, MySQL and Oracle). This should be safe, but may cause a crash if you attempt to use the schema_editor provided on these backends; in this case, pass atomic=False to the RunPython operation.

>If the RunPython operation is part of a non-atomic migration, the operation will only be executed in a transaction if atomic=True is passed to the RunPython operation.

The test suite only covered this behavior in the `apply` direction.  I've added a separate commit (5c9df6a) that augments the test suite to also test the behavior in the `unapply` direction.  I used a separate commit because they're really separate ideas (fixing a bug and increasing test coverage for an untested specified behavior). Please let me know if these should be squashed instead.